[CS2113-W12-3] MagusStock

ngkaiwen123 opened the pull request:
MagusStock is an inventory management system (IMS) that makes use of the command line interface for businesses to manage their stock inventory.

tsx0314 commented on docs/UML/Category/CategoryParser.puml:
The diagram is a bit hard to read. Perhaps you can omit some of the details

tsx0314 commented on docs/UML/History/HistoryStep1.png:
The square may be removed it should be "+" and "-"

tsx0314 commented on docs/UML/List/ListCommand.png:
Great! it is clear and easy to read!

tsx0314 commented on docs/UML/Alert/AlertStep1.puml:
need to replace the square with `-`

tsx0314 reviewed (commented):
Overall good job! Everything is clear and detailed! However, many explanations are used in the picture, sometimes it is hard to read

douph810975 commented on docs/SequenceDiagram.png:
A clear diagram here! I have a small question, why after the user calls the run function in MagusStock the MagusStock still call the run function to itself?

douph810975 commented on docs/ArchitectureDiagram.png:
This architecture diagram is clear and easy to understand!

douph810975 commented on docs/UML/Sell/SellCommand.png:
The picture is unable to display here. Are there any bugs here?

douph810975 commented on docs/UML/Restock/RestockCommand.png:
A meticulous diagram! I have a small problem, why after the updateItemQuality funciton, a new activation bar is created?

douph810975 reviewed (commented):
In general, the diagrams are clear and comprehensive.

chungnicholas commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The documentation and organisation of our project follows the recommended format as shown in [SE-Education](http://se-education.org/addressbook-level3/DeveloperGuide.html)
```
Consider having a full stop at the end of the sentence

chungnicholas commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### 1.1. Architecture Design Diagram
+![Architecture Diagram](ArchitectureDiagram.png)
```
Nice architecture design diagram

lil1n commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+specific pattern. If the input doesn't match the pattern, it prints an error message and returns. If the input matches 
+the pattern, it creates a new `Item` object using the parsed parameters and creates a new `AddCommand` object using the 
+`Inventory` object and `Item` object. It then calls the `run()` method of the `AddCommand` object to execute the add command.
+![AddParser.png](UML%2FAdd%2FAddParser.png)
```
I think that there is an additional return arrow between `AddParser` and `AddCommand`?

lil1n replied:
not sure if it is intended, but activation bar for `AddParser` seems to not be deactivated

chungnicholas commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+![ListCommand.png](UML%2FList%2FListCommand.png)
+
+
+**Step 1**. When the user executes the command `list`, the `ParserHandler` will create a new `ListParser` object and pass to it the `Inventory` where the items to be listed are stored.
+
+![ListStep1.png](UML%2FList%2FListStep1.png)
+
+**Step 2**. The `run` method in `ListParser` overrides the `run` method in `Parser` to create a new `ListCommand` object, passing to it the relevant `Inventory`.
+
+![ListStep2.png](UML%2FList%2FListStep2.png)
+
+**Step 3**. The `run` method in `ListCommand` overrides the `run` method in `Command` and calls the `listItems` method. The `listItems` method checks if the inventory is empty. If the inventory is empty, the method prints a message to inform the user that there are no items in the inventory. Otherwise, the `printTable` method from the `Ui` object is called.
+
+![ListStep3.png](UML%2FList%2FListStep3.png)
+
+**Step 4.**. If the `printTable` method is called, it takes in an `ArrayList<Item> items` aas a parameter and prints out a table showing the name, UPC, quantity and price of all items in the inventory.
+
```
Step by step guide is very clear and accompanied by diagrams for better visualization. Very nice!

chungnicholas replied:
Use of non-standard visibility notation.

lil1n commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+If the `SessionManager`'s `autoSave` flag is enabled, it writes the current inventory to a file using the `SessionManager`.
+
+The `run()` method of the `AddCommand` class calls the `addItem()` method to execute the add command.
+![AddCommand.png](UML%2FAdd%2FAddCommand.png)
```
not sure if it is intended, but I think that there is a missing activaction bar when `AddCommand` calls `addItems()` and also missing return arrow from `addItem()`?

lil1n replied:
not sure if it is intended, but activation bar for `AddCommand` seems to not be deactivated

lil1n replied:
not sure if `add(itemName, item)`, `writeSession(inventory)`, `writeCSV(inventory)` is correct, but I think that method calls should be `methodName(parameterName : parameterType)`?

chungnicholas commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
 # Developer Guide
+## Contents
+- [Acknowledgements](#acknowledgements)
+- [Design & Implementation](#design--implementation)
+- [Implementation](#implementation)
+  - [List](#list)
+  - [Add](#add)
+  - [Edit](#edit)
+  - [Remove](#remove)
+  - [Search](#search)
+  - [Filter](#filter)
+  - [Alert](#alert)
+  - [History](#history)
+  - [Category](#category)
+- [Product Scope](#product-scope)
+  - [Target User Profile](#target-user-profile)
+  - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Glossary](#glossary)
+- [Instructions for Manual Testing](#instructions-for-manual-testing)
 
```
Neat table of content. Maybe could consider adding numbering here as well, corresponding to those used in the headers so the reader can navigate easier when scrolling.

For example, this could be `2.3. Edit`
![image](https://user-images.githubusercontent.com/84615041/228703518-bf5de21f-813d-499f-9ef3-86a1ee61e99a.png)
corresponding to this 
![image](https://user-images.githubusercontent.com/84615041/228703655-60719d89-0cfc-40fb-8c0d-93c394af001d.png)

chungnicholas reviewed (commented):
The Developer Guide is well done and formatting is neat. Good job!

lil1n commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+
+Included below is a sequence diagram for the `RestockParser` and `RestockCommand` respectively:
+
+![RestockParser.png](UML/Restock/RestockParser.png)
```
not sure if it is intended, but diagram for `:RestockParser`, `:RestockCommand`, `:SellParser` and `:SellCommand` seems to be inconsistent with the rest of the diagrams as other diagrams do not have a `:` before the `XYZParser` and `XYZCommand`

lil1n commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+Included below is a sequence diagram for the `SellParser` and the `SellCommand` respectively:
+
+![SellParser.png](UML/Sell/SellParser.png)
+![SellCommand.png](UML/Sell/SellCommand.png)
```
there seems to be a lot of self calls, maybe can abstract out?

lil1n reviewed (commented):
Sequence diagrams are very detailed, but there are some areas that can be improved, great effort overall

chungnicholas commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+
+### 1.2. UML Sequence Diagram
+
+![Sequence Diagram](SequenceDiagram.png)
```
Classes have a missing `:` at the start. For example, `:MagusStock`. 

Activation bar for `MagusStock` is also missing a closing line. 
![image](https://user-images.githubusercontent.com/84615041/228706370-7a9ef577-3f79-4046-953c-c95cdff765da.png)

chungnicholas commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,406 @@
+### 2.1. List
+The list command is mainly handled by the `ListCommand` class, which extends the `Command` class.
+
+![ListCommand.png](UML%2FList%2FListCommand.png)
```
The `alt` frame is missing an `else` path.

chungnicholas replied:
Missing separate sequence diagram to show details of `ref` frames.

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,407 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### 1.1. Architecture Design Diagram
+![Architecture Diagram](ArchitectureDiagram.png)
```
Clear architecture design

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,407 @@
 # Developer Guide
+## Contents
+- [Acknowledgements](#acknowledgements)
+- [Design & Implementation](#design--implementation)
+- [Implementation](#implementation)
+  - [List](#list)
+  - [Add](#add)
+  - [Edit](#edit)
+  - [Remove](#remove)
+  - [Search](#search)
+  - [Filter](#filter)
+  - [Alert](#alert)
+  - [History](#history)
+  - [Category](#category)
+- [Product Scope](#product-scope)
+  - [Target User Profile](#target-user-profile)
+  - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
+- [Non-Functional Requirements](#non-functional-requirements)
+- [Glossary](#glossary)
+- [Instructions for Manual Testing](#instructions-for-manual-testing)
 
```
Clean and consise.

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,407 @@
+
+Breakdown of each component and its role in the application:
+
+`MagusStock`: This is the entry point for the application, and it's responsible for starting the application and coordinating the interactions between the other components.
+
+`ParserHandler`: This component is responsible for handling user input and determining which parser to execute based on the input. It uses a Parser to parse the input and generate a corresponding Command object.
+
+`Parser`: This component is responsible for parsing the user input and generating a Command object. The ParserHandler uses the Parser to parse the input and determine which Command object to create.
+
+`SessionManager`: This component is responsible for managing the inventory data persistence. It's connected to the Storage component, which reads and writes inventory data to and from a CSV file.
+
+`Storage`: This component is responsible for reading and writing inventory data to a CSV file. It's connected to the SessionManager, which uses it to manage the inventory data persistence.
+
+`Ui`: This component is responsible for displaying information to the user. It receives messages and data from the other components and displays them to the user.
```
Great short summary of each component

HiIAmTzeKean commented on docs/DeveloperGuide.md:
```diff
@@ -1,12 +1,407 @@
+`Inventory` object and `Item` object. It then calls the `run()` method of the `AddCommand` object to execute the add command.
+![AddParser.png](UML%2FAdd%2FAddParser.png)
+
+#### 2.2.2. AddCommand Class
+The `AddCommand` class is a command class that extends the `Command` abstract class. It represents the command to add an 
+item to the inventory. It contains a constructor that takes in an `Inventory` object and an `Item` object as parameters. 
+The constructor sets the `Inventory` object to be the inventory of the `Command` class, and the `Item` object to be the item 
+to be added to the inventory. The class also contains a private method called `addItem()` that adds the `Item` object to 
+the inventory. The `addItem()` method checks if the item already exists in the inventory using its unique UPC 
+(Universal Product Code) code. If the item already exists, it prints a message stating that the item is a duplicate. 
+Otherwise, it adds the item to the inventory, updates the item name hash, and adds the item to the UPC code history. 
+If the `SessionManager`'s `autoSave` flag is enabled, it writes the current inventory to a file using the `SessionManager`.
+
```
I like the way the classes are elaborated in segments so that developers can search a certain segment they want to read more about.

HiIAmTzeKean reviewed (commented):
Overall it is a great DG!

Ng-YZ committed:
Modify filter category function and revert category (list/table)

Ng-YZ committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

firwer committed:
Merge pull request #203 from firwer/master

Enforcement of valid range & UG clarity

Ng-YZ committed:
Fix merge conflict

Ng-YZ committed:
Update category-related UG information

ngkaiwen123 committed:
User Guide reformatting for Sell/Restock/Dashboard Command.

Ng-YZ committed:
UG: Update sample outputs for anything printed in tables

ngkaiwen123 committed:
Merge branch 'master' into master

Ng-YZ committed:
Edit 'Uncategorized' to 'uncategorized' to pass tests

ngkaiwen123 committed:
no message

Ng-YZ committed:
Edit UG

ysl-28 committed:
Add printing of alert warnings when alerts are created

Ng-YZ committed:
Clean up category code

ngkaiwen123 committed:
-Edit command added OutfRange Checking Capability.

Ng-YZ committed:
Simplify remove function, edit junit tests with RemoveCommand used

ngkaiwen123 committed:
-Added assertions for edit command testing

Ng-YZ committed:
modify UG anchors

ysl-28 committed:
Merge pull request #213 from ysl-28/master

Add printing of alert warnings when alerts are created

Ng-YZ committed:
Simplify remove command further, remove redundant ui methods, add filter category test

Ng-YZ committed:
Fix merge conflict, clean up redundant ui variables

ngkaiwen123 committed:
-Updating PPP document

ysl-28 committed:
Update tables in UG to show index column

Ng-YZ committed:
Update help function's list of commands

Ng-YZ committed:
Append category table formatting

Ng-YZ committed:
Fix merge conflicts

Ng-YZ committed:
Checkstyle: align indentation

Ng-YZ committed:
Merge pull request #212 from Ng-YZ/master

Modify filter category and remove function

ngkaiwen123 committed:
-Editing Test Codes, further updates to user guide.

ngkaiwen123 committed:
Further updates to user guide

ngkaiwen123 committed:
Merge branch 'master' into master

ngkaiwen123 committed:
Merge pull request #194 from ngkaiwen123/master

Bug Fixes, User Guide Update, Added assertions to tests

ngkaiwen123 committed:
Further changes to user guide

ngkaiwen123 committed:
Merge pull request #214 from ngkaiwen123/master

Further fixes to user guide

ysl-28 committed:
Update tables in UG to include index column

ysl-28 committed:
Merge pull request #216 from ysl-28/master

Update tables in UG to include index column

ysl-28 committed:
Add handling of invalid input for list command

ysl-28 committed:
Merge pull request #217 from ysl-28/master

Add handling of invalid input for list command

ArtemiszenN committed:
Fix search message appearing randomly when loading the program.
Change some string literals to constants

ArtemiszenN committed:
checkstyle grrrr

ArtemiszenN committed:
Change filter command help table thing

ArtemiszenN committed:
Merge pull request #218 from ArtemiszenN/newMaster

Fix search message appearing randomly when loading the program.

firwer committed:
PPP, Invalid input standardisation and Minor UG edits

firwer committed:
PPP update

firwer committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ysl-28 committed:
Add maximum range enforcement for alert adding

ysl-28 committed:
Remove unused comment

ngkaiwen123 committed:
-Edits to the User Guide, Developer Guide

ngkaiwen123 committed:
Merge pull request #221 from ngkaiwen123/master

-Edits to the User Guide, Developer Guide

firwer committed:
refactor: name of PPP

firwer committed:
Merge branch 'master' of https://github.com/firwer/tp

ysl-28 committed:
Edit alert regex to reject extra whitespace

ysl-28 committed:
Edit warning message for OutOfRangeException

ysl-28 committed:
Update AddParserTest

firwer committed:
Merge branch 'master' into master

firwer committed:
Add support for punctuations to add command

firwer committed:
Merge branch 'master' of https://github.com/firwer/tp

firwer committed:
docs: Further improvement to UG

Ng-YZ committed:
Edit addWordWithoutWrap function to print sufficient spaces

Ng-YZ committed:
Append ui function to display '[]' in help table

Ng-YZ committed:
Merge pull request #222 from Ng-YZ/master

Fix ui printing for table (unique rows)

ysl-28 committed:
Fix typo in minimum alert warning and add more AddAlertCommand tests

ysl-28 committed:
Standardise printing of maximum quantity limit

ysl-28 committed:
Merge pull request #220 from ysl-28/master

Add maximum range enforcement for alert adding

firwer committed:
Merge branch 'master' into master

firwer committed:
Merge pull request #219 from firwer/master

UG Edit, Invalid Input standardisation and PPP

ArtemiszenN committed:
weipin why'd you merge when it can't compile ASKDLJHASKLJDH

ArtemiszenN committed:
Merge pull request #223 from ArtemiszenN/newMaster

weipin why'd you merge when it can't compile ASKDLJHASKLJDH

ysl-28 committed:
Add javadoc comments

ysl-28 committed:
Fix formatting of help table in UG

ysl-28 committed:
Resolve merge conflict

ysl-28 committed:
Merge pull request #224 from ysl-28/master

Add javadoc comments

ysl-28 committed:
Update PPP

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Merge pull request #231 from ysl-28/master

update PPP

ngkaiwen123 committed:
-Fixes to edit command
-No longer prints same category for showing of updates.
-Capital letters/Special characters now supported.

ngkaiwen123 committed:
Minor edits to DG

ngkaiwen123 committed:
Small revamp of edit command test

ngkaiwen123 committed:
Merge pull request #232 from ngkaiwen123/master

Bug Fixes For #225 and #226

ArtemiszenN committed:
Harden session csv
Synchronise csv status

ArtemiszenN committed:
Merge pull request #233 from ArtemiszenN/newMaster

Harden session csv

ArtemiszenN committed:
change readme to work with about us page
change benchmark data to fit within 500kb
change benchmark data to be txt to align with savefile extension

ArtemiszenN committed:
Merge pull request #234 from ArtemiszenN/newMaster

change readme to work with about us page

ArtemiszenN committed:
Move benchmark data to data

ArtemiszenN committed:
Merge pull request #235 from ArtemiszenN/newMaster

Move benchmark data to data

ArtemiszenN committed:
checkstyle

ArtemiszenN committed:
Merge pull request #236 from ArtemiszenN/newMaster

checkstyle

ngkaiwen123 committed:
Developer Guide Update for UML Diagrams for both Edit and Restock Commands.

ngkaiwen123 committed:
Merge branch 'master' of https://github.com/ngkaiwen123/tp

ngkaiwen123 committed:
Reedited the UML Diagrams for Developer Guide.

ngkaiwen123 committed:
Developer Guide edits

ngkaiwen123 committed:
Documentation again

ysl-28 committed:
Add more alert tests

ngkaiwen123 committed:
-More test cases to ensure edit command is 99.999999999999999999% bug free

ngkaiwen123 committed:
Merge pull request #237 from ngkaiwen123/master

Developer Guide edits, more test cases, PPP edits

ysl-28 committed:
Update UG with example of alert warning message

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Merge pull request #238 from ysl-28/master

Add more alert tests and update UG

ArtemiszenN committed:
Add ppp for arty

ArtemiszenN committed:
Merge remote-tracking branch 'upstream/master' into newMaster

ArtemiszenN committed:
Merge pull request #239 from ArtemiszenN/newMaster

arty ppp ppp ppp

ArtemiszenN committed:
ppp ppp ppp

ArtemiszenN committed:
Merge pull request #240 from ArtemiszenN/newMaster

ppp ppp ppp

firwer committed:
docs: DG improv

firwer committed:
merge conflict fix

ysl-28 committed:
Update command summary

ysl-28 committed:
Update PPP

firwer committed:
fix: checkstyle & lowercase for add func

firwer committed:
fix: checkstyle

firwer committed:
fix: checkstyle again

firwer committed:
merge conflict

Ng-YZ committed:
print category name with first letters of important words capitalised

Ng-YZ committed:
Edit function now updates categoryHash

Ng-YZ committed:
Remove debugging printing statements

Ng-YZ committed:
extract variables to be static final variables

ysl-28 committed:
Update AlertParser diagram

Ng-YZ committed:
Small touchups

Ng-YZ committed:
Fix checkstyle

Ng-YZ committed:
Modify tests' expected categories

Ng-YZ committed:
oops missed some expected outputs

Ng-YZ committed:
Merge pull request #243 from Ng-YZ/master

Fix category issues

ysl-28 committed:
Update Add/RemoveAlertCommand diagrams

Ng-YZ committed:
Add PPP, add tests

ysl-28 committed:
Update DG and alert diagrams

ysl-28 committed:
Update PPP

Ng-YZ committed:
Add on to PPP

Ng-YZ committed:
Merge pull request #244 from Ng-YZ/master

PPP

ysl-28 committed:
Resolve merge conflict

ysl-28 committed:
Merge pull request #245 from ysl-28/master

Update Alert sequence diagrams in DG

ArtemiszenN committed:
Add a new test to try and find the elusive remove bug

ArtemiszenN committed:
add category to the test

ArtemiszenN committed:
Merge pull request #246 from ArtemiszenN/newMaster

Add a new test to try and find the elusive remove bug

ArtemiszenN committed:
Harden csv for rare bug

ngkaiwen123 committed:
-Slight edits to user guide, fix category for edit command

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ArtemiszenN committed:
change ppp file name for artemis to match github name

ArtemiszenN committed:
Merge pull request #247 from ArtemiszenN/newMaster

Harden csv for rare upc bug

ngkaiwen123 committed:
Merge pull request #248 from ngkaiwen123/master

Slight edits to UG

firwer committed:
Merge branch 'master' into master

firwer committed:
fix: checkstyle

firwer committed:
Merge pull request #242 from firwer/master

DG: Improv

ArtemiszenN committed:
style thingies

ArtemiszenN committed:
Merge remote-tracking branch 'upstream/master' into newMaster

ngkaiwen123 committed:
-DG Fix

ArtemiszenN committed:
no message

ysl-28 committed:
Edit parser conditions to check for empty string

Ng-YZ committed:
Modify uml diagrams for remove and category

Ng-YZ committed:
Add line of code to UML diagrams to follow format on website

ngkaiwen123 committed:
Anchor links are now working

ngkaiwen123 committed:
Merge pull request #253 from ngkaiwen123/master

-DG and UG anchor Fix

Ng-YZ committed:
Add line of code and titles to rest of UML to standardise

Ng-YZ committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

Ng-YZ committed:
Forgot to add line of code to overall sd oops

Ng-YZ committed:
Update RemoveAlert uml 'coz opt block was misplaced :(

ysl-28 committed:
Add maximum qty value of 99999999 to Types

Ng-YZ committed:
Updated PPP

ysl-28 committed:
Standardise case of parameters in UG

Ng-YZ committed:
Missed a few other stuff, modified.

Ng-YZ committed:
Fix one typo

ysl-28 committed:
Edit UG with extremely minor standardisation

ysl-28 committed:
Edit UG with extremely minor standardisation

ysl-28 committed:
Resolve merge conflict

Ng-YZ committed:
Merge pull request #255 from Ng-YZ/master

DG: check UML diagrams and update accordingly

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Update sequence diagrams for ListCommand and AlertParser

ysl-28 committed:
Update images for sequence diagrams

ArtemiszenN committed:
Merge remote-tracking branch 'upstream/master' into newMaster

ArtemiszenN committed:
Merge pull request #252 from ArtemiszenN/newMaster

The great code cleanup 2023

ngkaiwen123 committed:
changed potential incorrect output for edit command, developer guide manual testing.

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ngkaiwen123 committed:
-More additions to DG Manual Testing Section. All that is left is alert and history.

ngkaiwen123 committed:
Edits to Filter section of manual testing.

ngkaiwen123 committed:
Merge pull request #258 from ngkaiwen123/master

DG Edits (Manual Testing Section)

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

Ng-YZ committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

Ng-YZ committed:
DG: add manual testing cases for alert, history and filter category

ngkaiwen123 committed:
-DG,PPP Edits

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

Ng-YZ committed:
Merge pull request #260 from Ng-YZ/master

DG: add manual testing info for some other functions

Ng-YZ committed:
Merge branch 'master' into master

Ng-YZ committed:
Update PPP

Ng-YZ committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Update alert UML diagrams

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Merge pull request #256 from ysl-28/master

Update sequence diagrams and edit parser conditions

ngkaiwen123 committed:
Merge pull request #259 from ngkaiwen123/master

DG,PPP Edits

Ng-YZ committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

Ng-YZ committed:
Update About us

Ng-YZ committed:
Merge pull request #261 from Ng-YZ/master

docs: Update About Us, PPP

ngkaiwen123 committed:
-More formatting fix to DG, ensure that website shows correct formatting.

ngkaiwen123 committed:
About Us edit

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ngkaiwen123 committed:
no message

ngkaiwen123 committed:
Merge branch 'master' of https://github.com/ngkaiwen123/tp

ArtemiszenN committed:
price / qty can now be 0 in save file

ngkaiwen123 committed:
Merge pull request #262 from ngkaiwen123/master

DG Formatting, UG Addition

ArtemiszenN committed:
Merge pull request #263 from ArtemiszenN/newMaster

price / qty can now be 0 in save file

ArtemiszenN committed:
page break dg
filter cateogry small letter

ArtemiszenN committed:
Merge pull request #264 from ArtemiszenN/newMaster

page break dg

ngkaiwen123 committed:
-Added UG QnA section

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ArtemiszenN committed:
filter lowercase

ArtemiszenN committed:
Merge pull request #266 from ArtemiszenN/newMaster

filter lowercase

ngkaiwen123 committed:
Merge pull request #265 from ngkaiwen123/master

UG QnA Section

Ng-YZ committed:
Update CategoryFormatException.java

Ng-YZ committed:
Update HistoryErrorException.java

Ng-YZ committed:
Update MissingParametersException.java

Ng-YZ committed:
Update SearchFilterErrorException.java

ngkaiwen123 committed:
-Fix to Cat

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ngkaiwen123 committed:
Merge pull request #268 from ngkaiwen123/master

Fix to Cat double printing

Ng-YZ committed:
Update MissingParametersException.java

ysl-28 committed:
Merge pull request #267 from Ng-YZ/master

Update exceptions printing

Ng-YZ committed:
Update EditCommand.java

Ng-YZ committed:
Merge branch 'master' into master

ArtemiszenN committed:
Integrate alert with storage

Ng-YZ committed:
Update CategoryCommand.java

ArtemiszenN committed:
Fix corrupt/empty alert file message

Ng-YZ committed:
Update EditCommand.java

ArtemiszenN committed:
fix typo in edit warning

ArtemiszenN committed:
Merge pull request #275 from ArtemiszenN/newMaster

Integrate alert with storage

ysl-28 committed:
Merge pull request #274 from Ng-YZ/master

Update EditCommand for duplicate printing of category bug

ArtemiszenN committed:
add add and alert sample formats

ArtemiszenN committed:
small new line fix

ArtemiszenN committed:
fix test

ArtemiszenN committed:
Merge pull request #277 from ArtemiszenN/newMaster

add add and alert sample formats

Ng-YZ committed:
DG: Edit UML diagram

ngkaiwen123 committed:
Update PPP

ngkaiwen123 committed:
Merge branch 'master' of https://github.com/ngkaiwen123/tp

Ng-YZ committed:
Merge pull request #278 from Ng-YZ/master

DG: Update diagrams and text

ngkaiwen123 committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

ArtemiszenN committed:
Sync db storage status with actual status

ngkaiwen123 committed:
Merge pull request #279 from ngkaiwen123/master

Update personal PPP

firwer committed:
DG Spacing

firwer committed:
Merge branch 'master' of https://github.com/firwer/tp

ArtemiszenN committed:
Merge pull request #280 from ArtemiszenN/newMaster

Sync db storage status with actual status

firwer committed:
DG update

firwer committed:
Merge branch 'AY2223S2-CS2113-W12-3:master' into master

firwer committed:
PPP addon

firwer committed:
PPP edit 2.0

firwer committed:
Merge pull request #281 from firwer/master

DG, UG, PPP Edits

firwer committed:
docs add profile pic

firwer committed:
Merge pull request #282 from firwer/master

docs add profile pic

ysl-28 committed:
Update PPP

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Merge pull request #283 from ysl-28/master

Update PPP

ysl-28 committed:
Update PPP

ysl-28 committed:
Update PPP

ysl-28 committed:
Fix checkstyle and javadoc

ysl-28 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W12-3/tp

ysl-28 committed:
Merge pull request #284 from ysl-28/master

Fix checkstyle and add javadoc
